File: app/src/main/java/me/toptas/rssreader/base/BasePresenter.java
package me.toptas.rssreader.base;

public class BasePresenter<V extends BaseView> implements BaseMvpPresenter<V> {

    /**
     * Attached view
     */
    private V mView;


    @Override
    public void attach(V view) {
        mView = view;
    }

    @Override
    public void detach() {
        mView = null;
    }

    @Override
    public boolean isAttached() {
        return mView != null;
    }

    public V getView() {
        return mView;
    }
}

File: app/src/main/java/me/toptas/rssreader/base/BasePresenter.kt
package me.toptas.rssreader.base

open class BasePresenter<V : BaseView> : BaseMvpPresenter<V> {

    /**
     * Attached view
     */
    var view: V? = null


    override val isAttached = view != null


    override fun attach(view: V) {
        this.view = view
    }

    override fun detach() {
        view = null
    }
}